python/nano/src/bigdl/nano/tf/keras/Model.py

```diff
@@ -13,11 +13,92 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-
-
+from time import time
 import tensorflow as tf
+import numpy as np
 
 
 class Model(tf.keras.Model):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
+
+    def fit(self,
+            x=None,
+            y=None,
+            batch_size=None,
+            epochs=1,
+            verbose='auto',
+            callbacks=None,
+            validation_split=0.,
+            validation_data=None,
+            shuffle=True,
+            class_weight=None,
+            sample_weight=None,
+            initial_epoch=0,
+            steps_per_epoch=None,
+            validation_steps=None,
+            validation_batch_size=None,
+            validation_freq=1,
+            max_queue_size=10,
+            workers=1,
+            use_multiprocessing=False,
+            perf_tune=False):
+        if perf_tune:
+            if batch_size is None:
+                batch_size = 32
+            # Data pipeline
+            print("Prefetch Datapipeline")
+            assert issubclass(
+                type(x), tf.data.Dataset), "Data object currently only support tf.data.dataset"
+
+            batched_data=[]
+            batched_target = []
+            for i in range(batch_size):
```

Comment:
dataset is already assumed to be batched. see https://www.tensorflow.org/api_docs/python/tf/keras/Model#fit

Reply:
1. `dataset = dataset.take(1).cache().repeat(100)` cache the first elements and repeat to create enough data for 100 iterations, this step is to see the effects of dataset to throughput
2.  `dataset = dataset.unbatch().batch(new_batch_size).cache().repeat(100)`, compare to 1 to see the effects of batch_size on throughput